syntax = "proto3";
package localcontroller;

service LocalController {
  rpc StartTelemetry(StartTelemetryRequest) returns (StartResponse);
  rpc StartCallback(StartCallbackRequest) returns (StartResponse);
  rpc Configure(ConfigureRequest) returns (ConfigureResponse);
}

message StartTelemetryRequest {
  string kafka_broker_addr = 1;
  string topic = 2;
}

message StartCallbackRequest {
  string server_addr = 1;
}

message ConfigureRequest {
  string msg = 1;
}

message StartResponse {
  string status = 1;
  string message = 2;
}

message ConfigureResponse {
  string status = 1;
}